P0009/P0009.bs

```diff
@@ -1763,24 +1759,19 @@ Table �: Accessor requirements
   <th>Requirements/Notes</th>
 </tr>
 <tr>
-  <td>`A::value_type`</td>
-  <td></td>
-  <td></td>
-</tr>
-<tr>
-  <td>`A::pointer`</td>
+  <td>`A::reference<T>`</td>
   <td></td>
-  <td>*Requires:* `A::pointer` shall meet the requirements of random access iterator (**[random.access.iterators]**), and `iterator_traits<A::pointer>::value_type` shall be exactly `A::value_type`.</td>
+  <td>*Requires:* `A::reference<T>` shall be convertable to `T` and, if `!is_const_v<T>`, assignable from `T`. *[Note:* This may be `T&` or a proxy that operates like `T&`. *--end note]*</td>
 </tr>
 <tr>
-  <td>`A::reference`</td>
+  <td>`A::pointer<T>`</td>
   <td></td>
-  <td>*Requires:* `iterator_traits<A::pointer>::reference` shall be convertible to `A::reference` </td>
+  <td>*Requires:* `A::pointer<T>` shall be `DefaultConstructible` and `CopyAssignable`. *[Note:* This may be `T*` or a proxy that operates like `T*`. *--end note]*</td>
```

Comment:
This doesn't seem to be enough in terms of what pointer is. This for example doesn't define that operator[] (ptrdiff_t) exists or how that behaves. It doesn't define that operator* () exists or what it does. And we need arithmetic on those to allow subspan. If we define all of those than we are probably getting really close to random access iterator. Is there anything specific in random access iterator we don't want or shouldn't require for pointer?

Reply:
Operational spec required `p[i]` to return a reference to the offset - so that is still there.
An iterator also requires `*p`, `++p`, and `p++` - which should not be required by operational spec.

Reply:
The point about `*p`, `++p`, and `p++` not needing to exist is interesting. Do you envision any cases where this would be necessary? You're losing a lot of the power you get by separating storage from access, and it's not clear how you would form the codomain span with something that doesn't meet the requirements of `RandomAccessIterator`.